### DISCUSSION TITLE
Any ideas on migrating Anka templates to Tart?

### DISCUSSION THREAD

**Discussion by remarkov** [cirruslabs/tart#171] [General]
We have a heavily customized Anka template which would be nice to just migrate to Tart and avoid doing everything from scratch.
I totally understand that in most use cases this would be absolutely useless, especially if Packer is used for the VM setup but since both use the same Apple Virtualization I thought maybe there's a way to migrate a VM from Anka format to the one used by Tart? 
Looking at the file structure I doubt so but would like to hear what Tart devs and advanced users think.

**Comment by fkorotkov** [comment]
At Cirrus Labs we hesitant to even look into Anka or Anka format in order to not break proprietary something. It seems to be a grey area since it's not a reverse engineering of the software per se but only looking into the output format... But I'm not a lawyer and would check with Veertu Labs to get some written permit to do such conversion from their format into Tart's format. Or consult with a knowledgable lawyer to see if reverse engineering of the output format is a potential violation of something.

But I can tell you what `Virtualization.Framework` needs to run a VM and where it's stored in Tart's VM directory (`~/.tart/vms/VM_NAME`) so one can construct it manually or automatically from any source that uses `Virtualization.Framework`. 

To run a VM `Virtualization.Framework` uses 4 things:

1. [The main disk image](https://developer.apple.com/documentation/virtualization/VZDiskImageStorageDeviceAttachment) is located at `~/.tart/vms/VM_NAME/disk.img`
2. [Data used by the boot loader](https://developer.apple.com/documentation/virtualization/VZMacAuxiliaryStorage) is stored in `~/.tart/vms/VM_NAME/nvram.bin`.
3. `ecid` field in `~/.tart/vms/VM_NAME/config.json` is just a base64 representation of [`VZMacMachineIdentifier`](https://developer.apple.com/documentation/virtualization/VZMacMachineIdentifier).
4. `hardwareModel` field in `~/.tart/vms/VM_NAME/config.json` is just a base64 representation of [`VZMacHardwareModel`](https://developer.apple.com/documentation/virtualization/VZMacHardwareModel).

**Reply by remarkov** [reply]
Thank you Fedor, I totally understand and appreciate your point. Keep up the great work you do!
